### ISSUE TITLE
Authentication on refresh has to be explicit but shouldn't be

### ISSUE THREAD

**Issue by jsfan** [requests/requests-oauthlib#202]
[RFC 6749](https://tools.ietf.org/html/rfc6749#section-2.3)

**Comment by Lukasa** [comment]
@jsfan Do you have a proposed API for this?

**Comment by jsfan** [comment]
The approach I tried turned out not to work and I haven't had a chance to revisit and try something else. It has not been forgotten though.

**Comment by mmerickel** [comment]
Yes auto-refresh seems to be currently broken unless you pass the `client_secret` along with every request. Ideally this would be stored in the `OAuth2Session` itself so that it was there when it was needed.

**Comment by btimby** [comment]
My proposal is to accept `client_secret` in `__init__()` for use in `refresh_token()`. `client_id` is already passed and accessible.

https://github.com/requests/requests-oauthlib/issues/264
